### ISSUE TITLE
Add a pathway to rename Template Parts from the toolbar

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#38908]
Currently the only way to rename a template part is via the Advanced panel in the Inspector:

<img width="282" alt="Screenshot 2022-02-18 at 10 30 52" src="https://user-images.githubusercontent.com/846565/154665747-875a082a-8ee4-46f8-bd06-e9bc6971b69d.png">

It may be worth offering a pathway to do this from the block toolbar, for situations where the Inspector is closed. It's not a frequent flow, so the more menu seems like a good home for such an option:

<img width="701" alt="Screenshot 2022-02-18 at 10 28 17" src="https://user-images.githubusercontent.com/846565/154666164-6f9fb97d-c3f1-4298-b8e6-2b9207a6eee0.png">

Clicking "Rename" could invoke the same modal we currently use for naming template parts on creation:

<img width="370" alt="Screenshot 2022-02-18 at 10 29 21" src="https://user-images.githubusercontent.com/846565/154666381-5ac90dd7-5d96-4e94-9adf-e25fbda51669.png">

**Comment by jasmussen** [comment]
Realizing that the ´[Icon] Header` part of the toolbar currently opens a dropdown, my first instinct would be to have a rename feature in context of that label.

One thing we know we want to get to, is to refactor the transforms menu. As a place that can hold block, pattern, and style transformations all in one place, it could (should?) be one of the first things you check for a block when you want to explore alternate designs. 

If the transforms menu became more of a _destination_, would it provide a spot for renaming? What might it look like?

**Comment by jameskoster** [comment]
Yes I think that makes sense. All template part specific actions (swapping, editing, detaching) could potentially live there.

**Comment by strarsis** [comment]
+1, also a theme template part overridden by the user will keep the old template part name, even if the theme changes it (as in `theme.json`, `templateParts[]/name`).
